### ISSUE TITLE
Unclear how to handle modal interactions 

### ISSUE THREAD

**Issue by Chrispy404** [slack-go/slack#751]
While PR #724 gives a nice example of how to display a modal, it does nothing to show how to process data submitted in it. A modal is not displayed as a final step, it's displayed for interaction. 

This example feels like it's only half there and leaves users like me confused as to how to process the modal further.

**Comment by mkurde** [comment]
I second @Chrispy404 request and would like to ask @BattleBas shed some light on his example, add some documentation, or extend the example to be useful and ready to use for beginners. Please.

**Comment by sspaink** [comment]
@Chrispy404 and @mkurde I created a pull request with a more complete and usable example showing how to open a modal using a slash command and then using the user data to send a response.

Let me know if you need more clarifications or see a mistake.

**Comment by mkurde** [comment]
Thx, I am going to check that out. In the meantime, I have figured all out myself. Once your new PR is merged, I will check and add more examples if needed.
